File: ts-version/src/man.ts
class Man extends Warrior {

  protected static readonly STARTING_HEALTH: number = 55;
  protected readonly ATTACK_POW: number = 6;
  protected readonly ATTACK_PROB: number = 67;
  protected readonly CRITICAL_PROB: number = 0;
  protected readonly SHEILD_PROB: number = 17;

  protected readonly type: string = "Man";

  constructor(name: string = "Generic Man", index: number = 0, health: number = Man.STARTING_HEALTH, color: string = "White") {
    super(name, index, health, color)
  }

  getAttackPow(): number {
    return this.ATTACK_POW;
  }

  getAttackProb(): number {
    return this.ATTACK_PROB;
  }

  getCriticalProb(): number {
    return this.CRITICAL_PROB;
  }

  getSheildProb(): number {
    return this.SHEILD_PROB;
  }

  getType(): string {
    return this.type
  }

  battleCry(): string {
    return "Oh man...";
  }
}

class GondorMan extends Man {

  protected static readonly STARTING_HEALTH: number = 60;
  protected readonly ATTACK_POW: number = 5;
  protected readonly ATTACK_PROB: number = 67;
  protected readonly CRITICAL_PROB: number = 0;
  protected readonly SHEILD_PROB: number = 33;

  protected readonly type: string = "Gondor Man";

  constructor(name: string = "Generic Gondor Man", index: number = 0, health: number = GondorMan.STARTING_HEALTH, color: string = "White") {
    super(name, index, health, color)
  }

  getAttackPow(): number {
    return this.ATTACK_POW;
  }

  getAttackProb(): number {
    return this.ATTACK_PROB;
  }

  getCriticalProb(): number {
    return this.CRITICAL_PROB;
  }

  getSheildProb(): number {
    return this.SHEILD_PROB;
  }

  getType(): string {
    return this.type
  }

  battleCry(): string {
    return "Oh man...";
  }
}

class RohanMan extends Man {

  protected static readonly STARTING_HEALTH: number = 75;
  protected readonly ATTACK_POW: number = 4;
  protected readonly ATTACK_PROB: number = 83;
  protected readonly CRITICAL_PROB: number = 0;
  protected readonly SHEILD_PROB: number = 17;

  protected readonly type: string = "Rohan Man";

  constructor(name: string = "Generic Rohan Man", index: number = 0, health: number = RohanMan.STARTING_HEALTH, color: string = "White") {
    super(name, index, health, color)
  }

  getAttackPow(): number {
    return this.ATTACK_POW;
  }

  getAttackProb(): number {
    return this.ATTACK_PROB;
  }

  getCriticalProb(): number {
    return this.CRITICAL_PROB;
  }

  getSheildProb(): number {
    return this.SHEILD_PROB;
  }

  getType(): string {
    return this.type
  }

  battleCry(): string {
    return "In Rohan, we ride horses and fight!";
  }
}

File: src/Man.java
public class Man extends Warrior {

  private static final int STARTING_HEALTH = 55;
  private static final int ATTACK_POW = 6;
  private static final int ATTACK_PROB = 67;
  private static final int CRITICAL_PROB = 0;
  private static final int SHEILD_PROB = 17;

  public static final String type = "Man";

  public Man() {
    super("Generic Man", 0, STARTING_HEALTH);
  }

  public Man(int i) {
    super("Generic Man", i, STARTING_HEALTH);
  }

  public Man(int i, boolean isUser) {
    super("Generic Man", i, STARTING_HEALTH, isUser);
  }

  public Man(String name) {
    super(name, 0, STARTING_HEALTH);
  }

  public Man(String name, boolean isUser) {
    super(name, 0, STARTING_HEALTH, isUser);
  }

  public Man(String name, int i) {
    super(name, i, STARTING_HEALTH);
  }

  public Man(String name, int i, boolean isUser) {
    super(name, i, STARTING_HEALTH, isUser);
  }

  public Man(String name, int i, int health) {
    super(name, i, health);
  }

  public Man(String name, int i, int health, boolean isUser) {
    super(name, i, health, isUser);
  }

  public int getAttackPow() {
    return ATTACK_POW;
  }

  public int getAttackProb() {
    return ATTACK_PROB;
  }

  public int getCriticalProb() {
    return CRITICAL_PROB;
  }

  public int getSheildProb() {
    return SHEILD_PROB;
  }

  public String getType() {
    return type;
  }

  public String battleCry() {
    return "Oh man...";
  }

}
